spec/adapter/mysql_spec.cr

```diff
@@ -99,6 +99,7 @@ describe Granite::Adapter::Mysql do
   Spec.before_each do
```

Comment:
Unfortunately, this is a dangerous think to do. `Spec.before_each` will run before every single test in the test suite. It is not scoped by file nor `describe` block.

Often that leads to unintended consequences down the road.

That said, these statements look like they would do no harm, and if they would for a future test, there would likely be a compile time error.

Reply:
Right, we are aware if that limitation.  Do you have a better recommendation than doing it this way?

Reply:
For this code, I think it is fine - I don't recommend or expect any changes.

I'm more wary of it in generated code - as the end user may not know the implications of putting things in there.

Any solution I have isn't perfect. You can wrap the code you want in a macro, and insert that macro "before_each" test where it applies.

Reply:
This specific pattern was copied straight out of the other tests, but I'm [aware that it's not ideal](https://github.com/crystal-lang/crystal/issues/5094). I toyed around with some other patterns, but it ultimately nothing cleaned up the pattern.

Reply:
One thought I had is that, should this pattern propagate and be used in a greater portion of the test suite, it may be useful to create a set of models which are used throughout the test suite. Dropped/created in spec_helper, and truncated as here but with the definition of that truncation clearly placed in spec_helper as well. 

Hopefully it would be easier to deal with the subtleties of database drivers which have different levels of adherence to the SQL spec in different ways, and that polymorphism would be constrained to the spec_helper as well (or perhaps in a kind of run-at-spec-start migration file for each driver).

As for the heavy use of macros, there may be a better way to write code and I'm open to suggestions. I think it's really important that a project like this have as much parity between drivers as possible. One good way to enforce that is through uniform testing. As features are added, it becomes more and more likely driver tests would diverge in subtle ways.